### ISSUE TITLE
Inspecting dictionary with double byte string fails

### ISSUE THREAD

**Issue by ericwinger** [ericwinger/Jade#651]
Inspection still uses old Jade code which doesn't handle multi-byte strings when inspecting dictionaries. At minimum, we should still bring up an inspector without error but show that the string was unprintable. 

Use case - put a double byte string into either the key or value of an entry in UserGlobals and try to inspect. Get walkback below.

The minimal fix should go into `JadeServer>>inspectDictionary:on:`

```
9:56:28 AM, Tuesday, December 3, 2019: 'Invalid format for Number'
Number class>>errorInvalidFormat
Number class>>fromString:
String>>asNumber
JadeNavigationInspector(JadeInspector)>>displayObject
JadeNavigationInspector(JadeInspector)>>onViewOpened
JadeNavigationInspector>>onViewOpened
JadeNavigationInspector(Shell)>>view:
JadeNavigationInspector(Shell)>>createView:
JadeNavigationInspector(JadeInspector)>>createView:
JadeInspector class>>showOn:session:
CodeSourcePresenter>>jadeInspect
Symbol>>forwardTo:
CommandDescription>>performAgainst:
[] in Command>>value
BlockClosure>>ifCurtailed:
BlockClosure>>ensure:
Command>>value
ShellView>>performCommand:
JadeWorkspace(Shell)>>performCommand:
CommandQuery>>perform
DelegatingCommandPolicy(CommandPolicy)>>route:
[] in ShellView(View)>>onCommand:
BlockClosure>>ifCurtailed:
BlockClosure>>ensure:
Cursor>>showWhile:
ShellView(View)>>onCommand:
ShellView(View)>>wmCommand:wParam:lParam:
ShellView(View)>>dispatchMessage:wParam:lParam:
[] in InputState>>wndProc:message:wParam:lParam:cookie:
BlockClosure>>ifCurtailed:
InputState>>wndProc:message:wParam:lParam:cookie:
ShellView>>translateAccelerator:
ShellView>>preTranslateKeyboardInput:
ShellView(View)>>preTranslateMessage:
InputState>>preTranslateMessage:
InputState>>pumpMessage:
InputState>>loopWhile:
InputState>>mainLoop
[] in InputState>>forkMain
ExceptionHandler(ExceptionHandlerAbstract)>>markAndTry
[] in ExceptionHandler(ExceptionHandlerAbstract)>>try:
BlockClosure>>ifCurtailed:
BlockClosure>>ensure:
ExceptionHandler(ExceptionHandlerAbstract)>>try:
BlockClosure>>on:do:
[] in BlockClosure>>newProcess
```

**Comment by ericwinger** [comment]
To add & remove a unicode symbol to UserGlobals, see `RowanServicesTest new`, `unicode method` category.

**Comment by ericwinger** [comment]
Here's a tenative fix/workaround. Full unicode support on client & server is the ultimate answer. 
```
JadeServer>>inspectDictionary: aDictionary on: aStream

	| keys keyDict |
	keys := self keysForDictionary: aDictionary.
	keyDict := Dictionary new.
	keys do: [:each | 
		| key |
		key := each printString , '~' , (self oopOf: each) printString.
		key := key collect: [:char | char asciiValue < 32 ifTrue: [$?] ifFalse: [char]].
		keyDict
			at: key
			put: each.
	].
	keys size printOn: aStream.
	aStream lf.
	keyDict keys asSortedCollection do: [:each | 
		| index keyString key value valueString |
		index := each findLast: [:char | char = $~].
		keyString := each copyFrom: 1 to: index - 1.
		keyString charSize = 1 ifFalse:[
			keyString := '<<unprintable key. charSize > 1>>']. 
		key := keyDict at: each.
		value := aDictionary at: key. 
		valueString := (self printStringOf: value to: 10).
		valueString charSize = 1 ifFalse:[
			valueString := '<<unprintable value. charSize > 1>>']. 
		aStream nextPutAll: keyString , '->' , valueString; tab.
		self print: (self oopOf: value) on: aStream.
		aStream lf.
	].
	^aStream 
		lf; 
		contents.
```

**Comment by LisaAlmarode** [comment]
For the record, my testing code

```
| dbs d|
dbs := 'a', (String with: (Character withValue: 353)), 'b'.

"keys and values in symbol dictionaries"
Published at: ('AKey1', dbs) asSymbol put: 'abc'.  "problem inspecting Published"
RowanLoader at: ('AKey2') asSymbol put: ('AValue1', dbs).
RowanTools at: ('AKey3') asSymbol put: ('AValue2', dbs) asSymbol.

"keys and values in regular Dictionaries"

d := Dictionary new.
d at: #key put: ('a', dbs) asString.
UserGlobals at: #aDict1 put: d.

d := Dictionary new.
d at: #key put: ('b', dbs) asUnicodeString.
UserGlobals at: #aDict2 put: d.

d := Dictionary new.
d at: #key put: ('c', dbs) asSymbol.
UserGlobals at: #aDict3 put: d.

d := Dictionary new.
d at: ('d', dbs) asString put: 23.
UserGlobals at: #aDict4 put: d.  "problem inspecting"

d := Dictionary new.
d at: ('e', dbs) asUnicodeString put: 23.
UserGlobals at: #aDict5 put: d. "problem inspecting"

d := Dictionary new.
d at: ('f', dbs) asSymbol put: 23.
UserGlobals at: #aDict6 put: d. "problem inspecting"

System commit.
```

**Comment by LisaAlmarode** [comment]
With the proposed fix, my test cases error more gracefully.

**Comment by ericwinger** [comment]
Going to call this fixed in 3.0.92pre-release2. A couple of tests were added based on @LisaAlmarode code and pass.
